// ngx-mqtt/esm2022/lib/mqtt.module.mjs
import { NgModule, InjectionToken } from '@angular/core';
import * as i0 from "@angular/core";
export const MQTT_SERVICE_OPTIONS = {
    connectOnCreate: true,
    hostname: 'localhost',
    port: 1884,
    path: ''
};
export const MqttServiceConfig = new InjectionToken('NgxMqttServiceConfig');
export const MqttClientService = new InjectionToken('NgxMqttClientService');
class MqttModule {
    static forRoot(config, client) {
        return {
            ngModule: MqttModule,
            providers: [
                {
                    provide: MqttServiceConfig,
                    useValue: config
                },
                {
                    provide: MqttClientService,
                    useValue: client
                }
            ]
        };
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.0.3", ngImport: i0, type: MqttModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.0.3", ngImport: i0, type: MqttModule }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.0.3", ngImport: i0, type: MqttModule }); }
}
export { MqttModule };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.0.3", ngImport: i0, type: MqttModule, decorators: [{
            type: NgModule
        }] });
